orbstation/quirks/damned.dm

```diff
@@ -0,0 +1,128 @@
+	var/lightning_damage_multiplier
+	/// Message displayed to viewers when the lightning strike procs.
+	var/lightning_message
+
+/datum/element/holy_weapon/Attach(datum/target, lightning_chance = 50, lightning_damage_multiplier = 0.10, lightning_message = DEFAULT_LIGHTNING_MESSAGE)
+	. = ..()
+	if(!isitem(target))
+		return ELEMENT_INCOMPATIBLE
+	src.lightning_chance = lightning_chance
+	src.lightning_damage_multiplier = lightning_damage_multiplier
+	src.lightning_message = lightning_message
+	RegisterSignal(target, COMSIG_ITEM_ATTACK, PROC_REF(on_attack))
+
+/datum/element/holy_weapon/Detach(datum/target)
+	. = ..()
+	UnregisterSignal(target, list(COMSIG_ITEM_ATTACK))
+
+/datum/element/holy_weapon/proc/on_attack(datum/target, mob/living/victim, mob/living/attacker)
+	SIGNAL_HANDLER
+
+	if(!HAS_TRAIT(victim, TRAIT_DAMNED) || !prob(lightning_chance) || victim.stat == DEAD)
+		return
+
+	lightningbolt(victim, lightning_damage_multiplier)
+	var/built_lightning_message = replacetext(lightning_message, "%VICTIM", victim)
+	victim.visible_message(built_lightning_message)
+	victim.apply_status_effect(/datum/status_effect/temporary_damned_halo)
+
+#undef DEFAULT_LIGHTNING_MESSAGE
+
+/obj/item/nullrod/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/holy_weapon)
+
+// Holy book blessing effect
+
+/obj/item/storage/book/bible/bless(mob/living/L, mob/living/user)
+	. = ..()
+
+	if(!ishuman(L))
+		return
+
+	var/mob/living/carbon/human/H = L
+	if(HAS_TRAIT(H, TRAIT_DAMNED))
+		H.visible_message(span_warning("[user] blinds [H] with the power of [deity_name]!"))
+		H.adjustFireLoss(5)
+		H.adjust_temp_blindness(5 SECONDS)
+		H.apply_status_effect(/datum/status_effect/temporary_damned_halo)
+		return FALSE
+
+	return
+
+// Holy water effect
+
+/datum/reagent/water/holywater/expose_mob(mob/living/exposed_mob, methods=TOUCH, reac_volume)
+	..() // this goes up here because otherwise the water will just extinguish you instantly
+	if(HAS_TRAIT(exposed_mob, TRAIT_DAMNED))
+		exposed_mob.set_wet_stacks(0)
+		exposed_mob.adjust_fire_stacks(3)
+		exposed_mob.ignite_mob()
+		exposed_mob.apply_status_effect(/datum/status_effect/temporary_damned_halo)
+
+/datum/reagent/water/holywater/on_mob_life(mob/living/carbon/affected_mob, delta_time, times_fired)
+	if(HAS_TRAIT(affected_mob, TRAIT_DAMNED))
+		affected_mob.set_confusion_if_lower(3 SECONDS)
+		affected_mob.adjustFireLoss(0.25*delta_time, 0)
+		affected_mob.adjustToxLoss(0.25*delta_time, 0)
+		affected_mob.apply_status_effect(/datum/status_effect/temporary_damned_halo)
+	return ..()
+
+// Holy explosion effect
+
+/datum/chemical_reaction/reagent_explosion/holyboom/on_reaction(datum/reagents/holder, datum/equilibrium/reaction, created_volume)
+	if(created_volume >= 50)
+		var/turf/T = get_turf(holder.my_atom)
+		var/effective_size = round(created_volume/20)
+		for(var/mob/living/carbon/C in get_hearers_in_view(effective_size,T))
```

Comment:
I am not that happy with looping over people a second time, but I guess it is fine.